### ISSUE TITLE
libfaad-2.0 fail to build with new gcc?

### ISSUE THREAD

**Issue by Randrianasulu** [heroineworshiper/hvirtual#3]
```

make[5]: Entering directory '/dev/shm/cin-HV/hvirtual/quicktime/thirdparty/faad2-2.0/libfaad'                                                 
  if /bin/sh ../libtool --mode=compile gcc -DHAVE_CONFIG_H -I. -I. -I..     -g -O2 -MT bits.lo -MD -MP -MF ".deps/bits.Tpo" -c -o bits.lo bits.c; \                                                                     
  then mv -f ".deps/bits.Tpo" ".deps/bits.Plo"; else rm -f ".deps/bits.Tpo"; exit 1; fi                                                           
 gcc -DHAVE_CONFIG_H -I. -I. -I.. -g -O2 -MT bits.lo -MD -MP -MF .deps/bits.Tpo -c bits.c  -fPIC -DPIC -o .libs/bits.o                          
In file included from common.h:347,   
   from bits.c:28:                 
 /usr/include/bits/mathcalls.h:317:1: error: conflicting types for 'lrintf'; have 'long int(float)' 
  317 | __MATHDECL (long int,lrint,, (_Mdouble_ __x));                        | ^~~~~~~~~~                                                 
     In file included from bits.c:28:
  common.h:313:23: note: previous definition of 'lrintf' with type 'int(float)'                                                                    
 313 |     static INLINE int lrintf(float f)                                 |                       ^~~~~~                                  
  make[5]: *** [Makefile:347: bits.lo] Error 1                        
  make[5]: Leaving directory '/dev/shm/cin-HV/hvirtual/quicktime/thirdparty/faad2-2.0/libfaad'                                                  
  make[4]: *** [Makefile:255: all-recursive] Error 1  
```
 

gcc 11.2.0 as in Slackware 15.0

is there possibility for upgrading this old copy of faad?

**Comment by Randrianasulu** [comment]
ah, commending out asm version of it fixed build


```
+++ b/quicktime/thirdparty/faad2-2.0/libfaad/common.h                   @@ -308,6 +308,7 @@ char *strchr(), *strrchr();                                  return i;                                                      
     }                                                                    
 #elif (defined(__i386__) && defined(__GNUC__))                    
   +   /*    #define HAS_LRINTF                                               
       // from http://www.stereopsis.com/FPU.html                              static INLINE int lrintf(float f)                                
  @@ -319,7 +320,7 @@ char *strchr(), *strrchr();       
   : "=m" (i)                                                           
   : "m" (f));                                                         
return i;                                                    
  -    }                                                              
    +    } */                                                                  
#endif

**Comment by Randrianasulu** [comment]
fixed in 91aaed7160f22359b1a62aad7a59f1ae58e804a1
